exonum/src/blockchain/mod.rs

```diff
@@ -168,6 +168,15 @@ impl Blockchain {
     pub fn service_keypair(&self) -> &(PublicKey, SecretKey) {
         &self.service_keypair
     }
+
+    /// Performs several shallow checks that transaction is correct.
+    ///
+    /// Returned `Ok(())` value doesn't necessarily mean that transaction is correct and will be
+    /// executed successfully, but returned `Err(..)` value means that this transaction is
+    /// **obviously** incorrect and should be declined as early as possible.
+    pub fn check_tx(snapshot: &dyn Snapshot, tx: &Verified<AnyTx>) -> Result<(), ExecutionError> {
```

Comment:
(Slightly) off-topic: I'm not sure that the API endpoint should return `ExecutionError` as is, especially given the recent work on unifying error format. Could be solved later, I guess.

Reply:
Yeah, I think we'll handle it within the API errors unification topic.